Tests/AccountRepositorySpec.swift

```diff
@@ -0,0 +1,283 @@
+                    expect(accountRepository.getUserIdentifiers()).to(elementsEqual(userInfoProvider.userIdentifiers))
+                }
+            }
+
+            context("when calling updateUserInfo()") {
+
+                it("will return false if no preference was set") {
+                    expect(accountRepository.updateUserInfo()).to(beFalse())
+                }
+
+                it("will return true if updateUserInfo() was called for the first time with empty preference") {
+                    accountRepository.setPreference(UserInfoProviderMock())
+                    expect(accountRepository.updateUserInfo()).to(beTrue())
+                }
+
+                it("will return true if updateUserInfo() was called for the first time with non-empty preference") {
+                    let userInfoProvider = UserInfoProviderMock()
+                    userInfoProvider.userID = "user"
+                    accountRepository.setPreference(userInfoProvider)
+                    expect(accountRepository.updateUserInfo()).to(beTrue())
+                }
+
+                it("will return false if no change was made") {
+                    accountRepository.setPreference(UserInfoProviderMock())
+                    accountRepository.updateUserInfo() // first call
+                    expect(accountRepository.updateUserInfo()).to(beFalse())
+                }
+
+                it("will return false if only accessToken was added") {
+                    let userInfoProvider = UserInfoProviderMock()
+                    accountRepository.setPreference(userInfoProvider)
+                    accountRepository.updateUserInfo() // first call
+                    userInfoProvider.accessToken = "token"
+                    expect(accountRepository.updateUserInfo()).to(beFalse())
+                }
+
+                it("will return false if only accessToken was changed") {
+                    let userInfoProvider = UserInfoProviderMock()
+                    userInfoProvider.userID = "user"
+                    userInfoProvider.accessToken = "tokenA"
+                    accountRepository.setPreference(userInfoProvider)
+                    accountRepository.updateUserInfo() // first call
+                    userInfoProvider.accessToken = "tokenB"
+                    expect(accountRepository.updateUserInfo()).to(beFalse())
+                }
+
+                it("will return true if userID was changed") {
+                    let userInfoProvider = UserInfoProviderMock()
+                    userInfoProvider.userID = "userA"
+                    accountRepository.setPreference(userInfoProvider)
+                    accountRepository.updateUserInfo() // first call
+                    userInfoProvider.userID = "userB"
+                    expect(accountRepository.updateUserInfo()).to(beTrue())
+                }
+
+                it("will return false if userID stayed the same changed") {
```

Comment:
what is "stayed the same changed" meant to be? "stayed the same" ?

Reply:
Haha, yes. Copy&paste leftover 😉